src/cme.jl

```diff
@@ -0,0 +1,76 @@
+
+"""
+Performs a breadth-first graph traversal of a  to enumerate all possible states.
+
+The truncation kwarg should be given as a vector of Int, denoting the maximum species' counts.
+If, upon taking traversing a reaction the amount of any species is greater than the value in the truncation vector, 
+the graph traversal will not add the state to the list of possible states.
+"""
+function traverse_reactionsystem(rs, u0; truncation=nothing)
+    num_molecules = sum(u0)
+    num_species = length(u0)
+    state_space = []
+    Q = Queue{typeof(u0)}()
+    enqueue!(Q, u0)
+    push!(state_space, u0)
+    nsm = netstoichmat(rs)
+    edges = eachcol(nsm)
+    while !isempty(Q)
+        w = dequeue!(Q)
+        for e in edges
+            x = w .+ e
+            if x ∉ state_space && all(y >= 0 for y in x)
+                if truncation === nothing 
+                    if sum(x) > num_molecules 
+                        error("need to provide truncation for infinite reaction system")
+                    else 
+                        push!(state_space, x)
+                        enqueue!(Q, x)
+                    end
+                else 
+                    if all(x[i] <= truncation[i] for i in 1:num_species)
+                        push!(state_space, x)
+                        enqueue!(Q, x)
+                    end
+                end
+            end
+        end
+    end
+    return state_space
+end
+
+function ODESystem_cme(rs, u0; truncation=nothing, kwargs...)
```

Comment:
If this approach is only going to generate the full CME for a specific, deterministic initial condition, that needs to be made clear in the documentation. I think in general a user would expect to get truncated equations which could be used for any initial condition (which is normally given by a vector of probabilities and does not require a deterministic number of each molecule initially).